### DISCUSSION TITLE
ampache CRON

### DISCUSSION THREAD

**Discussion by vontani** [ampache/ampache#3277] [Q&A]
Dear,

im trying to get cron up and runing, how ever, i did try to many times and cant get it work :/  get some errors out my comand line:
((( Server Ubuntu 20.4 ampache 5.3 )))

########################################
administrator@b2:~$ systemctl status ampache_cron
● ampache_cron.service - ampache_cron
     Loaded: loaded (/etc/systemd/system/ampache_cron.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Thu 2022-05-19 20:00:01 CEST; 2min 9s ago
TriggeredBy: ● ampache_cron.timer
       Docs: https://github.com/ampache/ampache/wiki
   Main PID: 433594 (code=exited, status=255/EXCEPTION)
########################################

followed by:
########################################
b2.domain.com php[433594]: PHP Fatal error:  Uncaught Ampache\Config\Init\Exception\EnvironmentNotSuitableException: Environment is not suitable - please refer to the installation guide in /home/USERNAME/domains/b2.domain.com/public_html/src/Config/Init/InitializationHandlerEnvironment.ph>
b2.domain.com php[433594]: Stack trace:
b2.domain.com php[433594]: #0 /home/USERNAME/domains/b2.domain.com/public_html/src/Config/Init/Init.php(58): Ampache\Config\Init\InitializationHandlerEnvironment->init()
b2.domain.com php[433594]: #1 /home/USERNAME/domains/b2.domain.com/public_html/src/Config/Init.php(37): Ampache\Config\Init\Init->init()
b2.domain.com php[433594]: #2 /home/USERNAME/domains/b2.domain.com/public_html/bin/cli(35): require('...')
b2.domain.com php[433594]: #3 {main}
b2.domain.com php[433594]:   thrown in /home/USERNAME/domains/b2.domain.com/public_html/src/Config/Init/InitializationHandlerEnvironment.php on line 43
b2.domain.com systemd[1]: ampache_cron.service: Main process exited, code=exited, status=255/EXCEPTION
b2.domain.com systemd[1]: ampache_cron.service: Failed with result 'exit-code'.
b2.domain.com systemd[1]: Failed to start ampache_cron.

AMPACHE_CRON.service
########################################
[Service]
PrivateTmp=true
KillMode=mixed
Type=oneshot
User=www-data  (or other ..)
Group=www-data  (or other ..)
ExecStart=php bin/cli run:cronProcess
WorkingDirectory=/home/USERNAME/domains/b2.domain.com/public_html
ProtectSystem=yes
########################################

ampache_cron.timer
ampache_cron.service
both are placed in systemd/system folder, ampache is setup and working, cron configs are don to.

any help / sugestion can help me out :)

**Comment by vontani** [comment]
some one can help ? sugestions ?

**Comment by lusum** [comment]
Can you activate the command of the cron process manually?

sudo -u www-data php bin/cli run:cronProcess

Btw those lines are wrong:

User=www-data (or other ..)
Group=www-data (or other ..)

correct one:

User=www-data
Group=www-data

Btw2:

WorkingDirectory=/home/USERNAME/domains/b2.domain.com/public_html

you have installed the server in your home directory, user and group should be the same of the web server

**Comment by vontani** [comment]
thanks! but i have explained my issue clear, your post just makes no sence at all, my working directory is my VPS and my user and group name are as shown " not www-data" some other 
how ever never mind, waiting so loong to get an clear post and nothing hapen pushes me just to an other instance,

**Comment by lusum** [comment]
I mean: it is to test if it is a cron issue or an ampache issue... Giving the fact that all other functions work.

**Comment by vontani** [comment]
PHP Fatal error: Uncaught Ampache\Config\Init\Exception\EnvironmentNotSuitableException: Environment is not suitable - please refer to the installation guide in "Environment is not suitable"

the cron dos not work, it just dont work, if you would like to help, make me know why or what this message means, there is no way that an cron just stuck with an message like " Environment is not suitable "

**Comment by lusum** [comment]
Ok, i checked the  code, problem is, that ampache don't give enough messages in case of this kind of failure, even turning debug to 5.
The best thing to do is to turn debug on and putting to 5 ( maximum level )

The code fails in a check function:

https://github.com/ampache/ampache/blob/b781147547a803b7b141372b6b823a423a336c9d/src/Module/Util/Environment.php#L37-L40

In the file 

https://github.com/ampache/ampache/blob/develop/src/Module/Util/Environment.php

you can find all the check done, it could be one of them.

Edit: opened an issue, with the target to give the user more info about the error.

**Comment by vontani** [comment]
i understande, check php / check dependecies check php_intl, if ampache checks that doing the installation, and everything looks fine, why cron results in an exit code <?
all modules are installed, is there some restrictions to watch out or ?

**Comment by lusum** [comment]
They only way is to execute the cron command in the command line, like i suggested.

sudo -u www-data php bin/cli run:cronProcess

If it fails, it is a configuration problem, otherwise a cron permission problem
